pkgs/development/python-modules/torchmetrics/default.nix

```diff
@@ -0,0 +1,81 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, cloudpickle
+, scikit-learn
+, scikitimage
+, packaging
+, psutil
+, py-deprecate
+, pytorch
+, pytestCheckHook
+, torchmetrics
+}:
+
+let
+  pname = "torchmetrics";
+  version = "0.8.1";
+in
+buildPythonPackage {
+  inherit pname version;
+
+  src = fetchFromGitHub {
+    owner = "PyTorchLightning";
+    repo = "metrics";
+    rev = "v${version}";
+    hash = "sha256-AryEhYAeC97dO2pgHoz0Y9F//DVdX6RfCa80gI56iz4=";
+  };
+
+  propagatedBuildInputs = [
+    packaging
+    py-deprecate
+  ];
+
+  # Let the user bring their own instance
+  buildInputs = [
+    pytorch
+  ];
+
+  checkInputs = [
+    scikit-learn
+    scikitimage
+    cloudpickle
+    psutil
+  ];
+
+  nativeBuildInputs = [
+    pytestCheckHook
+  ];
+
+  # A cyclic dependency in: integrations/test_lightning.py
+  doCheck = false;
+  dontUsePytestCheck = true;
+  passthru.tests.check = torchmetrics.overridePythonAttrs (a: {
+    doCheck = true;
+    dontUsePytestCheck = false;
```

Comment:
Isn't ``doCheck = false`` enough? I already applied this technique to multiple packages and needed never to set this.

Reply:
Now that I've moved the hook from native to check inputs - yes, it's enough